crates/bevy_utils/src/map.rs

```diff
@@ -38,6 +38,78 @@ impl<K: Hash + Eq + PartialEq + Clone, V> PreHashMapExt<K, V> for PreHashMap<K,
 /// Iteration order only depends on the order of insertions and deletions.
 pub type TypeIdMap<V> = HashMap<TypeId, V, NoOpHash>;
 
+/// Extension trait to make use of [`TypeIdMap`] more ergonomic.
+///
+/// # Examples
```

Comment:
I think this example is a bit too explicit :thinking: I would probably remove it, all methods are extremely simple.

Reply:
I think it’s worth having a simple example of what this is doing, but maybe mot one that uses all of the new functions - I’ll cut this down after adding `entry` and `entry_mut`.